src/plugins/graphql.js

```diff
@@ -46,28 +44,38 @@ function createWrapExecute (tracer, config, defaultFieldResolver, responsePathAs
 function createWrapParse (tracer, config) {
   return function wrapParse (parse) {
     return function parseWithTrace (source) {
-      const span = startSpan(tracer, config, 'parse')
-
-      analyticsSampler.sample(span, config.analytics)
+      let operationValue = ''
+      let span
 
       try {
         const document = parse.apply(this, arguments)
         const operation = getOperation(document)
 
         if (!operation) return document // skip schema parsing
+        operationValue = `${operation.operation}`
+        if (operation.name) {
+          operationValue = `.${operationValue}.${operation.name.value}`
+        } else {
+          operationValue = `.${operationValue}.${operation.selectionSet.selections[0].name.value}`
+        }
 
         Object.defineProperties(document, {
           _datadog_source: {
             value: source.body || source
           }
         })
+        span = startSpan(tracer, config, `parse${operationValue}`)
 
+        analyticsSampler.sample(span, config.analytics)
         addDocumentTags(span, document)
 
         finish(null, span)
 
         return document
       } catch (e) {
+        if (!span) {
```

Comment:
This signals to me that you've missed some of the logic of this function.  Why do you think the initial `startSpan` had to move?

Reply:
I'm starting it with operation information extracted from operation. This is what I set as the span name.

Reply:
You can call this and not have to duplicate span creation: https://github.com/opentracing/opentracing-javascript/blob/master/src/span.ts#L40